### ISSUE TITLE
Crashes on current bleed with enabled strict activity checking.

### ISSUE THREAD

**Issue by GraionDilach** [OpenRA/OpenRA#16498]
```
OpenRA engine version {DEV_VERSION}
Red Alert mod version {DEV_VERSION}
on map 6e75b9d1c9f341387e5fdc27843054857a79d55b (Alaska Anarchy Redux by Buddha).
Date: 2019-05-04 20:12:49Z
Operating System: Windows (Microsoft Windows NT 6.2.9200.0)
Runtime Version: .NET CLR 4.0.30319.42000
Exception of type `System.InvalidOperationException`: Actor mig 785 attempted to tick activity OpenRA.Mods.Common.Activities.FlyAttack after it had already completed.
   at OpenRA.Activities.Activity.TickOuter(Actor self) in d:\OpenRA-PRdev\OpenRA.Game\Activities\Activity.cs:line 54
   at OpenRA.Traits.ActivityUtils.RunActivity(Actor self, Activity act) in d:\OpenRA-PRdev\OpenRA.Game\Traits\ActivityUtils.cs:line 37
   at OpenRA.Actor.Tick() in d:\OpenRA-PRdev\OpenRA.Game\Actor.cs:line 144
   at OpenRA.World.Tick() in d:\OpenRA-PRdev\OpenRA.Game\World.cs:line 421
   at OpenRA.Game.InnerLogicTick(OrderManager orderManager) in d:\OpenRA-PRdev\OpenRA.Game\Game.cs:line 608
   at OpenRA.Game.LogicTick() in d:\OpenRA-PRdev\OpenRA.Game\Game.cs:line 632
   at OpenRA.Game.Loop() in d:\OpenRA-PRdev\OpenRA.Game\Game.cs:line 777
   at OpenRA.Game.Run() in d:\OpenRA-PRdev\OpenRA.Game\Game.cs:line 817
   at OpenRA.Game.InitializeAndRun(String[] args) in d:\OpenRA-PRdev\OpenRA.Game\Game.cs:line 255
   at OpenRA.Program.Main(String[] args) in d:\OpenRA-PRdev\OpenRA.Game\Support\Program.cs:line 37
```

[OpenRA-2019-05-04T200729Z.zip](https://github.com/OpenRA/OpenRA/files/3144789/OpenRA-2019-05-04T200729Z.zip)

**Comment by pchote** [comment]
Current bleed, and all earlier versions of bleed too. The strict activity check was added as a target to work towards, and while the work over the last few months has taken us much closer, its still not complete.

**Comment by GraionDilach** [comment]
Oh, I thought this was something which was working already and not an aim. NVM me then.

**Comment by pchote** [comment]
The specific crash here will/should be fixed by #16369.
